IdentityCore/src/network/request/MSIDUrlResponseSerializer.m

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDUrlResponseSerializer.h"
+#import "MSIDUrlResponse.h"
+
+@implementation MSIDUrlResponseSerializer
+
+- (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse
+                           data:(NSData *)data
+                        context:(id <MSIDRequestContext>)context
+                          error:(NSError **)error
+{
+    if (!httpResponse && !data)
```

Comment:
looking at this line again, I think we should only check for if (!httpResponse) because if we don't have any http response, but we have data, then it is invalid response.

Reply:
Agree. Addressed.